compiler/src/iree/compiler/Codegen/SPIRV/ConvertToSPIRVPass.cpp

```diff
@@ -235,7 +236,13 @@ struct HALInterfaceBindingSubspanConverter final
       return subspanOp.emitError()
              << "failed to convert SPIR-V type: " << resultType;
     }
-
+    assert(
```

Comment:
We should actually `return notifyMatchFailure("expected no or zero byte offset")` for such cases?